x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ClusterComputeHandler.java

```diff
@@ -209,8 +210,8 @@ void runComputeOnRemoteCluster(
                         List.of(),
                         configuration,
                         configuration.newFoldContext(),
-                        exchangeSource,
-                        exchangeSink
+                        exchangeSource::createExchangeSource,
+                        () -> exchangeSink.createExchangeSink(() -> {})
```

Comment:
It seems all prod implementations are passing a noop runnable. Am I missing the actual usage?

Reply:
I will be added later in "retry node requests on shard-level failures" work.